### PR TITLE
Create endpoint for product page

### PR DESCRIPTION
**Pull request by TaranBohdan**

### CODE REVIEW

File: src/main/java/com/stamp/shop/product/entity/ProductDTO.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private Long id;
+    private String scu;
+    private String name;
+    private int numberOfReviews;
+    private BigDecimal price;
+    private int yearOfIssue;
+    private String countryOfIssue;
+    private String series;
+    private ProductCategory category;
+    private ProductDescription productDescription;
```
Comment (obaibula) [inline src/main/java/com/stamp/shop/product/entity/ProductDTO.java:24]: Create a separate record, and name it ProductDescriptionDTO. Don't mess entites and embeddables with dtos.

File: src/main/java/com/stamp/shop/product/entity/ProductDTO.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.stamp.shop.product.entity;
```
Comment (obaibula) [inline src/main/java/com/stamp/shop/product/entity/ProductDTO.java:1]: move to product.dto package

**Review by obaibula — CHANGES_REQUESTED** [review]

File: src/main/java/com/stamp/shop/product/entity/ProductDTO.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+public class ProductDTO {
```
[the remark this answers is not in the document]
Comment (TaranBohdan) [reply src/main/java/com/stamp/shop/product/entity/ProductDTO.java:14]: I made it as record, but I could not use it in Tests

File: src/main/java/com/stamp/shop/product/ProductService.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+public class ProductService {
+
+    private final ProductRepository productRepository;
+    private final ProductMapper productMapper;
+
+    public ProductDTO getById(Long id) {
+        var product = productRepository.findById(id);
+        return productMapper.toDTO(product
+                .orElseThrow(() -> new NotFoundProductException(String.format(
+                        "Product with id %s was not found", id
```
[the remark this answers is not in the document]
Comment (obaibula) [reply src/main/java/com/stamp/shop/product/ProductService.java:19]: Also it should be "%d" instead of "%s", because Long id is not String.

**Review by obaibula — APPROVED** [review]
Fix comments before merge

**Pull request merged**